### PR TITLE
[Coding Guidelines] Add meeting minutes 2024-12-04

### PR DESCRIPTION
**Pull request by vapdrs**
Feel free to suggest changes if I didn't properly capture anyone's proposals or questions

### CODE REVIEW

**Review by JoelMarcey — COMMENTED** [review]
Thank you for taking the minutes today. Leaving to @PLeVasseur to approve.

File: subcommittee/coding-guidelines/meetings/2024-December-04/minutes.md [hunk 1]
```diff
@@ -28,14 +48,88 @@ Supplemental material to the agenda can be found on the [GitHub repo](https://gi
 
 ## Tasks
 
-* 
+* [PL](mailto:plevasseur@gmail.com): Create a set of milestones for progress moving forward
```
Comment (PLeVasseur) [inline subcommittee/coding-guidelines/meetings/2024-December-04/minutes.md:51]: :saluting_face:

File: subcommittee/coding-guidelines/meetings/2024-December-04/minutes.md [hunk 2]
```diff
@@ -28,14 +48,88 @@ Supplemental material to the agenda can be found on the [GitHub repo](https://gi
...
+
+* JP: Different levels of assurance for different guidelines may be useful
+* AF: The definition of unsafe, means that the code will be unproven
+* JP: It should be clear in the guidelines that some unsafe should be used
+* JM: We should have specific recommendations with what to do with unsafe code
+* JJ: We shouldn’t just focus on the unsafe keyword
+* AF: Deadlocks and checked/unchecked arithmetic are examples of stuff outside of this
+* WP: Would this committee want to bless Cargo Vet or create recommendations for crates?
+* PL: Getting involved with Cargo Vet is a good idea
+* PL: There is a proposal to change the unsafe keyword
```
Comment (PLeVasseur) [inline subcommittee/coding-guidelines/meetings/2024-December-04/minutes.md:120]: In particular, there is a [thought](https://github.com/rust-lang/rfcs/pull/3484#issuecomment-2039501484) Niko had to try to distinguish when code should be marked as `trusted`, i.e. I, the programmer, have verified that although some guardrails are removed in the below section of code, we keep the car on the road.

Seems like a [Rust 2027 consideration](https://github.com/rust-lang/rfcs/pull/3484#issuecomment-2039501484) at this point.

You could include a little of this detail in something of an addendum with the links. What do you think?

**Review by PLeVasseur — COMMENTED** [review]
Hey @vapdrs -- thanks for capturing the minutes!

I put a couple of notes about potential, far from now changes, to the `unsafe` keyword I vaguely mentioned if you wanted to provide that context.

File: subcommittee/coding-guidelines/meetings/2024-December-04/minutes.md (hunk 2 above)
Comment (vapdrs) [reply subcommittee/coding-guidelines/meetings/2024-December-04/minutes.md]: I agree, I vaguely remember you mentioning more detail about that during the meeting that I couldn't capture in time. I have added the link and reformatted the statement here.

**Pull request merged**